### PR TITLE
Rename `tokio::select!`'s internal `util` module to `__tokio_select_util`

### PR DESCRIPTION
**Pull request by WilliamVenner**
## Motivation

This internal module's identifier can clash with existing identifiers in library users' code and produce confusing error messages.

```rust
mod util {
    pub fn hello_world() {
        println!("Hello, world!");
    }
}

#[tokio::main]
async fn main() {
    let f1 = async {};
    let f2 = async {};

    tokio::select! {
        _ = f1 => {
            // The following line would fix the compile error, showing that Tokio is creating an internal identifier called `util` which is clashing
            // use crate::util;

            util::hello_world();
        },

        _ = f2 => {}
    }
}
```

```rust
error[[E0425]](https://doc.rust-lang.org/stable/error-index.html#E0425): cannot find function `hello_world` in module `util`
  --> src/main.rs:14:19
   |
14 |             util::hello_world();
   |                   ^^^^^^^^^^^ not found in `util`
   |
help: consider importing this function
   |
1  | [use crate::util::hello_world;](https://play.rust-lang.org/#)
   |

For more information about this error, try `rustc --explain E0425`.
```

## Solution

I have renamed the `util` module in `tokio::select!` to `__tokio_util` to prevent any identifier clashes.

Additionally, I also added `#[doc(hidden)]` to the module.

### CODE REVIEW

File: tokio/src/macros/select.rs [hunk 1]
```diff
@@ -429,7 +429,8 @@ macro_rules! select {
         //
         // This module is defined within a scope and should not leak out of this
         // macro.
-        mod util {
+        #[doc(hidden)]
+        mod __tokio_util {
```
Comment (hawkw) [inline tokio/src/macros/select.rs:433]: take it or leave it: this is specifically for the `select` macro, so i might call it `__tokio_select_util`? not a big deal either way though.

**Review by hawkw — APPROVED** [review]
thanks for the fix, this seems right to me! i had one minor suggestion.

File: tokio/src/macros/select.rs (hunk 1 above)
Comment (WilliamVenner) [reply tokio/src/macros/select.rs]: Agreed - I've reset my PR to the current master with the doc fix you made to reduce clutter and made the requested change.

**Comment by Darksonn** [comment]
I'm fine with renaming the module, but honestly, we should just be removing these imports and use absolute paths.

**Pull request merged**